### PR TITLE
Do not fail when no metastore instances is set

### PR DESCRIPTION
**Pull request by yigress**
This fix covers the scenario when no hive-metastore-stance is specified, user may provide additional-cloud-sql-instances or not provide any instance at all, in such case we first check if additional-cloud-sql-instance is provided, if not, default to mysql for backward compatibility.

### CODE REVIEW

File: cloud-sql-proxy/cloud-sql-proxy.sh [hunk 1]
```diff
@@ -58,37 +58,49 @@ readonly USE_CLOUD_SQL_PRIVATE_IP
...
 
-# CLOUD SQL instance type is one of MYSQL, POSTGRES, SQL
-METASTORE_INSTANCE_TYPE=$(get_metastore_instance_type)
+# CLOUD SQL instance type is one of MYSQL, POSTGRES, SQLSERVER.
+# Type inferred in order of METASTORE_INSTANCE, ADDITIONAL_INSTANCES, default to MYSQL
+if [[ -n "${METASTORE_INSTANCE}" ]]; then
+  METASTORE_INSTANCE_TYPE=$(get_metastore_instance_type "${METASTORE_INSTANCE}")
+elif [[ -n "${ADDITIONAL_INSTANCES}" ]]; then
+  METASTORE_INSTANCE_TYPE=$(get_metastore_instance_type "${ADDITIONAL_INSTANCES}")
+fi
```
Comment (functicons) [inline cloud-sql-proxy/cloud-sql-proxy.sh:98]: As discussed offline, we should not get type from additional instances, instead add a metadata for type, if not provided, default to MYSQL.

**Review by functicons — CHANGES_REQUESTED** [review]

File: cloud-sql-proxy/cloud-sql-proxy.sh (hunk 1 above)
Comment (yigress) [reply cloud-sql-proxy/cloud-sql-proxy.sh:101]: added metadata cloud-sql-instance-type that takes priority

**Pull request merged**